test/eredis_tests.erl

```diff
@@ -10,10 +10,14 @@ multibulk_test_() ->
                    list_to_binary(create_multibulk(["SET", "foo", "bar"]))),
      ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$3\r\nbar\r\n">>,
                    list_to_binary(create_multibulk(['SET', foo, bar]))),
-
      ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$3\r\n123\r\n">>,
                    list_to_binary(create_multibulk(['SET', foo, 123]))),
 
-     ?_assertThrow({cannot_store_floats, 123.5},
-                   list_to_binary(create_multibulk(['SET', foo, 123.5])))
-    ].
+     %% Test floats formatting with io_lib:format.
+     ?_assertEqual(<<"*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$21\r\n123.50000000000000000\r\n">>,
+                   list_to_binary(create_multibulk(['SET', foo, 123.5]))),
```

Comment:
Always 20 digits? It means even a small number like 0.0 or 1.0 needs 21 bytes. Seems like a waste, but floats are not everywhere so perhaps it's acceptable. I'd prefer shorter version like "123.5" though...

Reply:
Totally agree on that and it makes sense that OTP-25 adds a new standard solution. I think it's a problem if the library provides different behavior when OTP-25 is used.

Would it be acceptable to use the internal `fwrite` call when OTP-25 is not available, and OTP-25 always when it is? That way, we don't need a conditional for the tests as they produce the same result - and never something unexpected, like postfix zeroes.

Reply:
:+1: Exactly this I just wrote in another comment. We only use `io_lib_format:fwrite_g/1` in old OTP versions, there is no risk it will break in the future.

Reply:
🏁 Done, should work for all of OTP < 21 and current + future versions